submissions/misshatsker/object-oriented-js/app.js

```diff
@@ -0,0 +1,194 @@
+const Character = function(x, y, speed, sprite, width, height) {
+    this.sprite = sprite;
+    this.speed = speed;
+    this.x = x;
+    this.y = y;
+    this.width = width;
+    this.height = height;
+};
+
+Character.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Enemy = function(x, y, speed, sprite, width, height) {
+    Character.call(this, x, y, speed, sprite, width, height);
+};
+
+Enemy.prototype.update = function(dt) {
+    const shift = this.speed * dt;
+    this.x = this.x + shift;
+    if (this.x >= rightSide) {
+        this.x = -this.width;
+    }
+};
+
+Object.setPrototypeOf(Enemy.prototype, Character.prototype);
+
+const Player = function(x, y, speed, sprite, width, height, enemies) {
+    Character.call(this, x, y, speed, sprite, width, height);
+    this.enemies = enemies;
+};
+
+Player.prototype.update = function() {
+    if (this.checkPlayerOverlapWithEnemies()) {
+        gameOver(false);
+    }
+}
+
+Player.prototype.handleInput = function(action) {
+    switch (action) {
+        case 'left': {
+            if (this.x > leftSide) {
+                this.x -= this.speed;
+            }
+            break;
+        }
+        case 'right': {
+            if (this.x + this.width + this.speed < rightSide) {
+                this.x += this.speed;
+            }
+            break;
+        }
+        case 'down': {
+            if (this.y + this.height + this.speed < bottomSide) {
+                this.y += this.speed;
+            }
+            break;
+        }
+        case 'up': {
+            this.y -= this.speed;
+            if (this.y <= 0) {
+                gameOver(true);
+            }
+
+            break;
+        }
+    }
+}
+
+Player.prototype.checkPlayerOverlapWithEnemies = function() {
+    return this.enemies.some((enemy) => {
+        return checkIsOverlap(
+            {
+                x: this.x,
+                y: this.y
+            },
+            {
+                x: this.x + this.width,
+                y: this.y + this.height
+            },
+            {
+                x: enemy.x,
+                y: enemy.y
+            },
+            {
+                x: enemy.x + enemy.width,
+                y: enemy.y + enemy.height
+            }
+        );
+    });
+}
+
+Player.prototype.setPosition = function(x, y) {
+    this.x = x;
+    this.y = y;
+}
+
+Object.setPrototypeOf(Player.prototype, Character.prototype);
+
+function getRandomNumberFromARange(min, max) {
+    return (Math.random() * (max - min)) + min;
+}
+
+function createRandomEnemy(row) {
+    const x = getRandomNumberFromARange(enemyMinX, enemyMaxX);
+    const y = initialGroundY + row * tileHeight;
+    const speed = getRandomNumberFromARange(enemyMinSpeed, enemyMaxSpeed);
+
+    return new Enemy(x, y, speed, enemySprite, enemyWidth, enemyHeight);
+}
+
+function createRandomEnemies(amount) {
+    return new Array(amount).fill(null).map((elem, i) => createRandomEnemy(i));
```

Comment:
There is a convention to use `_` for the `map` argument when it is not actually used, so `...map((_, i) => ...)`